test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,264 @@
+
+	spdxPath := filepath.Join("..", "fixtures", "golang-1.12-alpine.spdx.json")
+	reportPath := filepath.Join("..", "fixtures", "golang-1.12-alpine.sarif.json")
+
+	registryName := "test-registry"
+	registryURI := "ghcr.io"
+	registryRepository := "rancher-sandbox/sbombastic/test-assets/golang"
+	golangAlpineTag := "1.12-alpine"
+
+	pollInterval := 1 * time.Second
+	pollTimeout := 1 * time.Minute
+	var sbom storagev1alpha1.SBOM
+	var vulnReport storagev1alpha1.VulnerabilityReport
+	var image storagev1alpha1.Image
+	crName := "dfe56d8371e7df15a3dde25c33a78b84b79766de2ab5a5897032019c878b5932"
+
+	f := features.New("Registry CR Creation test").
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			manager := helm.New(cfg.KubeconfigFile())
+			err := manager.RunInstall(helm.WithName(releaseName),
+				helm.WithNamespace(cfg.Namespace()),
+				helm.WithChart("../../helm"),
+				helm.WithWait(),
+				helm.WithTimeout("3m"))
+
+			require.NoError(t, err, "sbombastic helm chart is not installed correctly")
+
+			err = storagev1alpha1.AddToScheme(cfg.Client().Resources(cfg.Namespace()).GetScheme())
+			require.NoError(t, err)
+
+			err = v1alpha1.AddToScheme(cfg.Client().Resources(cfg.Namespace()).GetScheme())
+			require.NoError(t, err)
+
+			return ctx
+		}).
+		Assess("Create Registry CR", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			registry := &v1alpha1.Registry{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      registryName,
+					Namespace: cfg.Namespace(),
+				},
+				Spec: v1alpha1.RegistrySpec{
+					URI:          registryURI,
+					Repositories: []string{registryRepository},
+				},
+			}
+			err := cfg.Client().Resources(cfg.Namespace()).Create(ctx, registry)
+			require.NoError(t, err)
+
+			// Init the var for later test and owner reference delete check
+			image = storagev1alpha1.Image{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      crName,
+					Namespace: cfg.Namespace(),
+				},
+			}
+			sbom = storagev1alpha1.SBOM{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      crName,
+					Namespace: cfg.Namespace(),
+				},
+			}
+			vulnReport = storagev1alpha1.VulnerabilityReport{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      crName,
+					Namespace: cfg.Namespace(),
+				},
+			}
+
+			return ctx
+		}).
+		Assess("SPDX SBOM is created with expected content", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			err := wait.For(
+				conditions.New(cfg.Client().Resources(cfg.Namespace())).ResourceMatch(
+					&sbom,
+					func(obj k8s.Object) bool {
+						sbomObj, ok := obj.(*storagev1alpha1.SBOM)
+						if !ok {
+							t.Fatal("unexpected type assertion failure")
+						}
+
+						return EqualReference(
+							sbomObj.Spec.ImageMetadata,
+							registryURI,
+							registryRepository,
+							golangAlpineTag,
+						)
+					}),
+				wait.WithInterval(pollInterval),
+				wait.WithTimeout(pollTimeout),
+			)
```

Comment:
I understand that I probably have a similar code in my e2e tests lol, but generally if your SBOM resource is created with all information filled and it will not be changed later, you can just wait for the resource to be created and check its content. This way, in case of failure, you don't have to wait for the whole timeout to fail the test.  

```suggestion
			var sbomObj *storagev1alpha1.SBOM
			err := wait.For(
				conditions.New(cfg.Client().Resources(cfg.Namespace())).ResourceMatch(
					&sbom,
					func(obj k8s.Object) bool {
						var ok bool
						sbomObj, ok = obj.(*storagev1alpha1.SBOM)
						if !ok {
							t.Fatal("unexpected type assertion failure")
						}
						return true
					}),
				wait.WithInterval(pollInterval),
				wait.WithTimeout(pollTimeout),
			)
			require.NoError(t, err)
			assert.True(t, EqualReference(
				sbomObj.Spec.ImageMetadata,
				registryURI,
				registryRepository,
				golangAlpineTag,
			))
			require.NoError(t, err, "SBOM CR did not reach expected state within %s", pollTimeout)
```